Python/Product/PythonTools/ptvsd_launcher.py

```diff
@@ -68,14 +69,60 @@
 # fix sys.path to be the script file dir
 sys.path[0] = ''
 
+def parse_version(version):
+    """Version format is expected to be <int>.<int>.<int><optional str><optional int>"""
+    try:
+        regex = r"([0-9]+)\.([0-9]+)\.([0-9]+)([a-zA-Z]*)([0-9]*)"
+        match = re.match(regex, version.upper(), re.IGNORECASE)
+        parser = [int, int, int, str, int]
+        default = [0, 0, 0, 'z', 0]
+        result = []
+        for n in range(0, len(match.groups())):
+            m = n + 1
+            v = match.group(m)
+            result.append(parser[n](v) if v else default[n])
+        return result
+    except Exception:
+        pass
+    return []
+
+def get_bundled_packages_path():
+    return os.path.join(os.path.dirname(__file__), 'Packages')
+
+def get_bundled_ptvsd_version():
+    try:
+        ptvsd_path = os.path.join(get_bundled_packages_path(), 'ptvsd', '__init__.py')
+        with open(ptvsd_path, 'r') as f:
+            lines = f.readlines()
+        for line in lines:
+            if line.startswith('__version__'):
+                _, version = (s.strip().strip('"') for s in line.split('='))
```

Comment:
You should also strip single quotes, in case it ever gets changed in ptvsd from "..." to '...'